### PR TITLE
first pass at filtering out unpublished data, as well as unwanted par…

### PR DESCRIPTION
**Pull request by ssoper-usgs**
…ameters

Before making a pull request
----------------------------

- [ ] Make sure all tests run
- [ ] Update the changelog appropriately

Title
-----------
Brief description of changes. Reference the JIRA ticket if appropriate

Description
-----------
If no JIRA ticket is referenced, describe the changes made. Note anything that you want the reviewers to know while
reviewing your pull request

After making a pull request
---------------------------
- [ ] If appropriate, put the link to the PR in the JIRA ticket
- [ ] Assign someone to review unless the change is trivial

### CODE REVIEW

File: src/test/resources/testResult/happyPath/discrete_ground_water_old.csv [hunk 1]
```diff
@@ -0,0 +1,10 @@
...
+"8BDA141822744BA5E0530100007FD075","393215104490001","2019-06-20 16:34:00","2019-06-20 16:40:01","EIN, EN, GJ, GJS",,"clear, hot, no precip, very light wind","true","{\"LevelsPerformed\": false, \"CollectionAgency\": \"USGS\", \"OtherSampleTaken\": false, \"SedimentSampleTaken\": false, \"BiologicalSampleTaken\": false, \"RecorderDataCollected\": false, \"WaterQualitySampleTaken\": false, \"GroundWaterLevelPerformed\": true, \"SafetyInspectionPerformed\": false}","2020-05-11 18:35:35.067498","Water level, depth LSD","GW level, calib steel tape","{\"Display\": \"120.28\", \"Numeric\": 120.28}","ft","{}","ReferencePrimary","Unspecified","Unspecified","Unspecified (GWSI Discrete)","2019-06-20 16:36:00",null,"false","true",null,"false",null,null,"{\"Cut\": {}, \"Hold\": {}, \"WaterLevel\": {}, \"TapeCorrection\": {}}"
```
Comment (ssoper-usgs) [inline src/test/resources/testResult/happyPath/discrete_ground_water_old.csv]: This file will not be staying.  However, it shows that at least the publish filter is working...

File: src/main/resources/sql/insertDiscreteGroundWaterData.sql [hunk 2]
```diff
@@ -29,37 +29,44 @@ insert into discrete_ground_water (
...
+    ,use_location_datum_as_reference
+    ,reading_qualifier
+    ,reading_qualifiers
+    ,ground_water_measurement
 
 from field_visit_header_info
-		 join field_visit_readings
-			  on field_visit_header_info.json_data_id = field_visit_readings.json_data_id
-			  and field_visit_header_info.field_visit_identifier = field_visit_readings.field_visit_identifier
+         left join field_visit_readings
```
Comment (dsteinich) [inline src/main/resources/sql/insertDiscreteGroundWaterData.sql]: leave off the lefts. without data in all the tables we should have no output.

File: src/main/resources/sql/insertDiscreteGroundWaterData.sql [hunk 3]
```diff
@@ -29,37 +29,44 @@ insert into discrete_ground_water (
...
+    ,location_identifier
+    ,start_time
+    ,end_time
+    ,party
+    ,remarks
+    ,weather
+    ,field_visit_header_info.is_valid is_valid_header_info
+    ,completed_work
+    ,last_modified
+    ,field_visit_readings.parameter
```
Comment (dsteinich) [inline src/main/resources/sql/insertDiscreteGroundWaterData.sql:43]: We should add the parm_cd (from aq_to_nwis_parm) to this table for use in the load lambda.

File: src/test/resources/testData/data_type_mapping.csv [hunk 4]
```diff
@@ -0,0 +1,18 @@
+parm_cd,stat_cd,data_type
```
Comment (dsteinich) [inline src/test/resources/testData/data_type_mapping.csv:1]: Per conversations on 6/9/202, 61055 appears to be the only valid one. The rest could be removed (should not be in the real table, can remain here if add value to the tests).

**Review by dsteinich — CHANGES_REQUESTED** [review]

File: src/main/resources/sql/insertDiscreteGroundWaterData.sql [hunk 5]
```diff
@@ -29,37 +30,44 @@ insert into discrete_ground_water (
...
+    ,aq_to_nwis_parm.parm_cd
```
Comment (ssoper-usgs) [inline src/main/resources/sql/insertDiscreteGroundWaterData.sql:44]: Added parm_cd with latest commit here, and on line 13.

**Review by dsteinich — APPROVED** [review]

**Pull request merged**